### DISCUSSION TITLE
Statically Linked Nodejs - How does one statically link dynamic libraries?

### DISCUSSION THREAD

**Discussion by CMCDragonkai** [yao-pkg/pkg#12] [General]
Suppose you have created a node app that uses pkg.

According to the README:

> Note that fully static Node binaries are not capable of loading native bindings, so you may not use Node bindings with linuxstatic.

That makes sense. But then, I have the actual shared object and the source code for building the nodejs shared object. How exactly would I bundle that into the statically linked nodejs so that it still works?

**Comment by robertsLando** [comment]
> How exactly would I bundle that into the statically linked nodejs so that it still works?

With static binaries, you can't. You have to use non-static binaries. If your target is armv7 I have some pre-built binaries available here: https://github.com/yao-pkg/pkg-binaries
